packer/plugin.go

```diff
@@ -151,12 +155,15 @@ func (c *PluginConfig) discoverExternalComponents(path string) error {
 	}
 	for pluginName, pluginPath := range pluginPaths {
 		newPath := pluginPath // this needs to be stored in a new variable for the func below
-		c.Provisioners.Set(pluginName, c.Client(newPath).Provisioner)
+		c.Provisioners.Set(pluginName, func() (packersdk.Provisioner, error) {
```

Comment:
Do we need this change for the data sources too? I'm not sure about that because the PR title says that is for plugins with old names, and those plugins should never have a data source.

Reply:
Yes !

Reply:
( in the (maybe unlikely) case someone builds a single plugin binary )

Reply:
Nice! I will open a PR for this.

Reply:
Awesome!!!